src/java/org/apache/cassandra/net/RequestCallbacks.java

```diff
@@ -351,13 +355,29 @@ public Mutation mutation()
             return getMutation(mutation);
         }
 
+        /**
+         * Used for sensors tracking. A safe alternative to {@link #mutation()} to access counter mutations without
+         * changing existing assert behavior.
+         */
+        public IMutation getIMutation()
```

Comment:
It looks weird, but the existing mutation() method returns Mutation which is convenient for hints. In order not to changes the assertion behavior and refactoring by returning `IMutation` - I opted in for a different getter. Note that `CounterMutation` is `IMutation` but not `Mutation` and is needed for sensors (but not hints)